test/ech_test.c

```diff
@@ -1032,12 +1131,496 @@ static int ech_test_file_read(int run)
     return rv;
 }
 
+/* calls with bad, NULL, and simple, arguments, for generic code coverage  */
+static int ech_api_basic_calls(void)
+{
+    int rv = 0;
+    SSL_CTX *ctx = NULL;
+    SSL *s = NULL;
+    OSSL_ECHSTORE *es = NULL, *es1 = NULL;
+    char *rinner, *inner = "inner.example.com";
+    char *router, *outer = "example.com";
+    unsigned char alpns[] = { 'h', '2' };
+    size_t alpns_len = sizeof(alpns);
+    char *gsuite = "X25519,hkdf-sha256,aes-256-gcm";
+    uint16_t gtype = 0xfe09;
+    unsigned char *rc = NULL;
+    size_t rclen = 0;
+    BIO *in = NULL;
+
+    /* NULL args */
+    if (!TEST_int_eq(SSL_CTX_set1_echstore(NULL, NULL), 0)) {
+        TEST_info("SSL_CTX_set1_echstore unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(SSL_set1_echstore(NULL, NULL), 0)) {
+        TEST_info("SSL_set1_echstore unexpected non-zero");
+        goto end;
+    }
+    es1 = SSL_CTX_get1_echstore(NULL);
+    if (!TEST_ptr_eq(es1, NULL)) {
+        TEST_info("SSL_CTX_get1_echstore unexpected result");
+        goto end;
+    }
+    es1 = SSL_get1_echstore(NULL);
+    if (!TEST_ptr_eq(es1, NULL)) {
+        TEST_info("SSL_get1_echstore unexpected result");
```

Comment:
this one is useful, but with a better message like "SSL_get1_echstore(NULL) failed ");

Reply:
Now I'm confused:-) Can you say why you think this one more useful?

Reply:
if you had written : 
 `````C
    if (!TEST_ptr((es1 = SSL_get1_echstore(NULL)))  /* do not write this, unless you are sure there is no side effect b.c of C macro */
````
On failure, the macro will have dump the origine of `es1` content.
So, in this case, TEST_info is adding a few more information.

Reply:
ah, I see - I think that's now overtaken by events (given the previous commit) but let me know if you disagree